src/utility/SuggestionsManager.ts

```diff
@@ -16,86 +18,162 @@ export abstract class SuggestionsManager {
+    private static _suggestions2: Object = { string: Array<string>()};
 
     /**
      * Initializes the suggestions repository.
      */
     public static initialize() {
-        SuggestionsManager.readsuggestionsFile();
+       SuggestionsManager.readsuggestionsFile();
     }
 
     /**
      * Adds a suggestion to the file and saves the file.
      * @param suggestion The suggestions to add.
      */
-    public static addsuggestion(suggestion: string): boolean | undefined {
-        let listSize: number = SuggestionsManager._suggestions.length;
-        for (let x: number = 0; x < listSize; x++) {
-            if (suggestion.toLowerCase() == SuggestionsManager._suggestions[x].toLowerCase())
-                return false;
+    public static addsuggestion(args: string): number | undefined {
+        let index = args.indexOf("/");
+        let tempList = args.substring(0, index).toLowerCase().trim();
+        let tempSuggestion = args.substring(index + 1).toLowerCase().trim();
+
+        if (!SuggestionsManager._suggestions.has(tempList))
+            return resultEnum.LISTDOESNOTEXIST;
+
+        let currentSuggestions = SuggestionsManager._suggestions.get(tempList);
+
+        if (currentSuggestions) {
+            for (let index = 0; index < currentSuggestions.length; index++) {
+                if (currentSuggestions[index] == tempSuggestion) {
+                    return resultEnum.SUGGESTIONEXISTS;
+                }
+            }
+            currentSuggestions.push(tempSuggestion);
+            SuggestionsManager._suggestions.set(tempList, currentSuggestions);
+            this.updateFile();
+            return resultEnum.SUGGESTIONADDED;
         }
-        SuggestionsManager._suggestions[this._suggestions.length] = suggestion;
-        fs.appendFileSync(ActBotConstants.suggestionsPath, "~\n" + suggestion);
-        return true;
+        else {
+            return resultEnum.DERP; // This shouldn't happen..
+        }
+    }
+
+    public static addList(list: string): number | undefined {
+        let tempList = list.toLowerCase().trim();
+        if ( SuggestionsManager._suggestions.has(tempList))
+            return resultEnum.LISTEXISTS;
+        SuggestionsManager._suggestions.set(tempList, new Array<string>());
+        this.updateFile();
+        return resultEnum.LISTADDED;
     }
 
     /**
      * Removes a suggestion from the list and updates the file.
      * @param suggestion The suggestion to remove.
      */
-    public static removesuggestion(suggestion: string): boolean | undefined {
-        let listSize: number = SuggestionsManager._suggestions.length;
-        for (let x: number = 0; x < listSize; x++) {
-            if (suggestion.toLowerCase() == SuggestionsManager._suggestions[x].toLowerCase()) {
-                SuggestionsManager._suggestions.splice(x, 1);
-                this.updateFile();
-                return true;
+    public static removesuggestion(args: string): number | undefined {
```

Comment:
Nitpick:

camelCase make this removeSuggestion

Reply:
camelCased